File: backend/meetup/user/src/main/java/org/zil/user/AppConfig.java
package org.zil.user;

import lombok.RequiredArgsConstructor;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;

@Configuration
@RequiredArgsConstructor
public class AppConfig {
    private final UserRepo userRepo;
    @Bean
    public UserDetailsService userDetailsService() {
        return username -> userRepo.findByEmail(username).orElseThrow( () -> new UsernameNotFoundException("User credentials not found"));
    }

}

File: backend/user.registration/src/main/kotlin/com/meet/user/registration/config/AppConfig.kt
package com.meet.user.registration.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.support.beans
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder

@Configuration
class AppConfig {

    @Bean
    fun bcrypt():BCryptPasswordEncoder = BCryptPasswordEncoder()

}